diagram.pu

```diff
@@ -0,0 +1,72 @@
+@startuml
+
+class UserHandler{  
+    service UserService
+}
+
+
+class DefaultUserService{
+    repository Repository
+}
+
+interface Service {
+    AuthenticateUser(ctx context.Context, email string, hash string) (bool, error)
+	CreateUser(ctx context.Context, user entities.User) error
+	UpdateUser(ctx context.Context, user entities.User) error
+	GetUser(ctx context.Context, userID string) (entities.User, error)
+	GetAllUsers(ctx context.Context) ([]entities.User, error)
+	DeleteUser(ctx context.Context, userID string) error
+	BulkCreateUser(ctx context.Context, users []entities.User) error
+	SetUserParents(ctx context.Context, userID string, parents []entities.User) error
+}
+
+class ProxyRepository{
+}
+
+interface Repository {
+    Authenticate(email string, hash string) (bool, error)
```

Comment:
Do you receive a hashed password or receive a string and hash it after that? It is only a naming issue, I know.

Reply:
its just a dummie  auth, I get and email and any string  and comapre with db